Loop/Models/RemoteCommand.swift

```diff
@@ -8,78 +8,34 @@
-            case .invalidCarbs:
-                return NSLocalizedString("Invalid carb amount", comment: "Remote command error description: invalid carb amount.")
-            }
-        }
-    }
-}
-
-
-enum RemoteCommand {
-    case temporaryScheduleOverride(TemporaryScheduleOverride)
-    case cancelTemporaryOverride
-    case bolusEntry(Double)
-    case carbsEntry(NewCarbEntry)
-}
-
 
 // Push Notifications
-extension RemoteCommand {
-    static func createRemoteCommand(notification: [String: Any], allowedPresets: [TemporaryScheduleOverridePreset], defaultAbsorptionTime: TimeInterval, nowDate: Date = Date()) -> Result<RemoteCommand, RemoteCommandParseError> {
+extension RemoteAction {
+    static func createRemoteAction(notification: [String: Any]) -> Result<RemoteAction, RemoteCommandParseError> {
         if let overrideName = notification["override-name"] as? String,
-            let preset = allowedPresets.first(where: { $0.name == overrideName }),
             let remoteAddress = notification["remote-address"] as? String
         {
-            var override = preset.createOverride(enactTrigger: .remote(remoteAddress))
+            var overrideTime: TimeInterval? = nil
             if let overrideDurationMinutes = notification["override-duration-minutes"] as? Double {
-                override.duration = .finite(TimeInterval(minutes: overrideDurationMinutes))
+                overrideTime = TimeInterval(minutes: overrideDurationMinutes)
             }
-            return .success(.temporaryScheduleOverride(override))
-        } else if let _ = notification["cancel-temporary-override"] as? String {
-            return .success(.cancelTemporaryOverride)
+            return .success(.temporaryScheduleOverride(RemoteOverrideAction(name: overrideName, durationTime: overrideTime, remoteAddress: remoteAddress)))
+        } else if let _ = notification["cancel-temporary-override"] as? String,
+                  let remoteAddress = notification["remote-address"] as? String
+        {
+            return .success(.cancelTemporaryOverride(RemoteOverrideCancelAction(remoteAddress: remoteAddress)))
         }  else if let bolusValue = notification["bolus-entry"] as? Double {
-            return .success(.bolusEntry(bolusValue))
+            return .success(.bolusEntry(RemoteBolusAction(amountInUnits: bolusValue)))
         } else if let carbsValue = notification["carbs-entry"] as? Double {
-
-            let minAbsorptionTime = TimeInterval(hours: 0.5)
-            let maxAbsorptionTime = LoopConstants.maxCarbAbsorptionTime
             
-            var absorptionTime = defaultAbsorptionTime
+            var absorptionTime: TimeInterval? = nil
             if let absorptionOverrideInHours = notification["absorption-time"] as? Double {
                 absorptionTime = TimeInterval(hours: absorptionOverrideInHours)
             }
             
```

Comment:
Carb food type was missing from the APIs. This will require a change to the Nightscout codebase in order to support this but should be ok when empty too.